lib/std/core.kk

```diff
@@ -72,6 +72,12 @@ pub alias io-total = <ndet,console,net,fsys,ui,st<global>>
 // The `:io-noexn` effect is used for functions that perform arbitrary I/O operations, but raise no exceptions
 pub alias io-noexn = <div,io-total>
 
+// The `:event-loop` effect signifies that a function requires an initialized event loop
+pub type event-loop :: X
```

Comment:
Here is one major change. Adding event loop to the core effects. 
The async library requires that there be an event loop around it, or at the very least a timer that can call callbacks.

In principle we could just use an implicit parameter for the handler for the async effect, to get the event loop and wrap it.
The problem with that approach is that if you use any libuv apis directly in callback style without the async library then you have to manually remember to initialize the event loop. 
We could in principle make a custom error and detect uninitialized event loops at runtime, but that would introduce runtime overhead for each libuv call that uses callbacks, so I opted to create this new effect instead.